src/OneExplorer/OneExplorer.ts

```diff
@@ -163,6 +163,50 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
     this._onDidChangeTreeData.fire();
   }
 
+  // TODO: Add move()
+
+  /**
+   * Rename a file of all types of nodes (baseModel, derivedModel, config) excepts for directory.
+   * It only alters the file name, not the path.
+   */
+  rename(oneNode: OneNode): void {
+    // TODO: prohibit special characters for security ('..', '*', etc)
+    let warningMessage;
+    if (oneNode.node.type === NodeType.baseModel) {
+      // TODO automatically change the corresponding files
+      warningMessage = 'WARNING: You may need to change input paths in the following cfg files:\n';
+      for (const conf of oneNode.node.childNodes) {
+        warningMessage += `${conf.name} \n`;
+      }
+    } else {
+      warningMessage = 'WARNING: Renaming may result in some unexpected changes on the tree view.';
+    }
+
+    vscode.window
+        .showInputBox({title: warningMessage, placeHolder: 'Enter(proceed) / Escape(cancel)'})
+        .then((answer) => {
+          if (answer === undefined) {
+            // User has pressed esc to cancel.
+            return;
+          } else {
+            return vscode.window.showInputBox({
+              title: 'Enter a file name:',
+              placeHolder: `${path.basename(oneNode.node.uri.fsPath)}`
+            });
+          }
+        })
```

Comment:
I pulled this pr and tried to use the function. For me, I felt the warning step is a little bit annoying. 
![image](https://user-images.githubusercontent.com/50489513/170408004-b3b30f6a-95e0-4562-b337-5b9d1bbc5091.png)
I read the description and put a new name to the empty box and pressed enter. However, nothing happened but a new input box appeared and requests new file name again. How about notifying the warning message to user while getting new file name? (Then, the message should be shorter than now..)

```ts
    vscode.window
        .showInputBox({title: 'Enter a file name. ([WARNING] Relative paths between model and cfg could be broken)',
              placeHolder: `${path.basename(oneNode.node.uri.fsPath)}`})
        .then(newname => {
          if (newname) {
            const dirpath = path.dirname(oneNode.node.uri.fsPath);
            const newpath = `${dirpath}/${newname}`;
            vscode.workspace.fs.rename(oneNode.node.uri, vscode.Uri.file(newpath));

            this.refresh();
            return;
          }
        });
```

Reply:
No... It's just a ask box to notify the users. You push enter in the box and then the next box which takes the name will appear.

Reply:
I think the UI is making people confused. I have revised the behavior: please visit the gif link again and find the updated gif.

Reply:
I like updated gif procedure. It's much clearer than before. :)